### PR TITLE
hack: updates update-codgen.sh for go modules

### LINKED ISSUE #35: update-codegen.sh is broken
`update-codegen.sh` references k8s.io/code-generator in the vendor directory https://github.com/knative/observability/blob/94718bbb2999e84330d7de3b3e761d03c514e59f/hack/update-codegen.sh#L26, however the vendor directory does not have k8s.io/code-generator.

### PR DESCRIPTION
**Pull request by poy**
This CL updates the update-codgen.sh script to work with go modules
to capture and use the correct version of k8s.io/code-generator
(instead of git modules).

fixes #35

### CODE REVIEW

**Comment by poy** [comment]
The main file to focus on is `hack/update-codegen.sh`. The rest are from updating the k8s.io/code-generator.

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -18,14 +18,51 @@ set -o errexit
...
+        # non-semver
+        sha=$(echo $code_generator_version | cut -d '-' -f3)
+        git checkout $sha
+    else
+        # semver
+        git checkout $code_generator_version
+    fi
+
+    # Install deps
+    GOPATH=$temp_dir go get ./...
```
Comment (pianohacker) [inline hack/update-codegen.sh]: We had to change this to `go mod vendor` to get it working locally, as `GOFLAGS="-mod=vendor"` is set in the repo's .envrc.

Comment (poy) [reply hack/update-codegen.sh]: Ah, good deal. Updated.

**Comment by pianohacker** [comment]
/lgtm

Thanks for the fix!

**Comment by Benjamintf1** [comment]
image
It doesn't make sense. How can the tests have passed and not passed?

**Comment by poy** [comment]
@Benjamintf1 I don't see that prow has been told to test. Try running:

/test pull-knative-observability-integration-tests

**Pull request merged**